airflow/www/utils.py

```diff
@@ -228,7 +228,7 @@ def task_instance_link(attr):
     dag_id = attr.get('dag_id')
     task_id = attr.get('task_id')
     execution_date = attr.get('dag_run.execution_date') or attr.get('execution_date') or timezone.utcnow()
-    url = url_for('Airflow.task', dag_id=dag_id, task_id=task_id)
+    url = url_for('Airflow.task', dag_id=dag_id, task_id=task_id, execution_date=execution_date)
```

Comment:
```suggestion
    url = url_for('Airflow.task', dag_id=dag_id, task_id=task_id, execution_date=execution_date.isoformat())
```

See https://github.com/apache/airflow/commit/944dcfbb918050274fd3a1cc51d8fdf460ea2429#diff-9baa686d200246adbc21b33dc8e57a2cc395c53330b67d304949479db99eb191L229